tests/unit/test-cmd/test.build.js

```diff
@@ -64,301 +56,271 @@ describe('build', () => {
       artifactsDir: 'artifacts',
       ignoreFiles: ['**/*.log'],
     };
-    return build(params, {packageCreator, createFileFilter}).then(() => {
-      // ensure sourceDir, artifactsDir, ignoreFiles is used
-      sinon.assert.calledWithMatch(createFileFilter, params);
-      // ensure packageCreator received correct fileFilter
-      sinon.assert.calledWithMatch(packageCreator, {fileFilter});
-    });
+    await build(params, {packageCreator, createFileFilter});
+    // ensure sourceDir, artifactsDir, ignoreFiles is used
+    sinon.assert.calledWithMatch(createFileFilter, params);
+    // ensure packageCreator received correct fileFilter
+    sinon.assert.calledWithMatch(packageCreator, {fileFilter});
   });
 
-  it('gives the correct name to a localized extension', () => {
-    return withTempDir(
-      (tmpDir) =>
-        build({
-          sourceDir: fixturePath('minimal-localizable-web-ext'),
-          artifactsDir: tmpDir.path(),
-        })
-          .then((buildResult) => {
-            assert.match(buildResult.extensionPath,
-                         /name_of_the_extension-1\.0\.zip$/);
-            return buildResult.extensionPath;
-          })
-    );
-  });
-
-  it('handles repeating localization keys', () => {
-    return withTempDir(
-      (tmpDir) => {
-        const messageFileName = path.join(tmpDir.path(), 'messages.json');
-        fs.writeFileSync(messageFileName,
-          `{"extensionName": {
-              "message": "example extension",
-              "description": "example description"
-            }
-          }`);
-
-        const manifestWithRepeatingPattern = {
-          name: '__MSG_extensionName__ __MSG_extensionName__',
-          version: '0.0.1',
-        };
+  it('gives the correct name to a localized extension', () => withTempDir(
+    async (tmpDir) => {
+      const buildResult = await build({
+        sourceDir: fixturePath('minimal-localizable-web-ext'),
+        artifactsDir: tmpDir.path(),
+      });
+      assert.match(buildResult.extensionPath,
+                   /name_of_the_extension-1\.0\.zip$/);
+      await buildResult.extensionPath;
+    }
+  ));
 
-        return getDefaultLocalizedName({
-          messageFile: messageFileName,
-          manifestData: manifestWithRepeatingPattern,
-        })
-          .then((result) => {
-            assert.match(result, /example extension example extension/);
-          });
-      }
-    );
-  });
+  it('handles repeating localization keys', () => withTempDir(
+    async (tmpDir) => {
+      const messageFileName = path.join(tmpDir.path(), 'messages.json');
+      fs.writeFileSync(messageFileName,
```

Comment:
nit, we could also rewrite this using `await fs.writeFile(...)`, instead of the sync version of the node fs method.